schema.tf

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) 2020 Oracle and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+# Tenancy Classification
+variable "class" {
+  type        = string
+  description = "The tenancy classification sets boundaries for resource deployments"
+  default     = "PAYG"
+}
+# Resident Configuration
+variable "parent" {
+  type = string
+  description = "The Oracle Cloud Identifier (OCID) for a parent compartment, an encapsulating child compartment will be created to define the service resident. Usually this is the root compartment, hence the tenancy OCID."
+}
+variable "organization"            { 
+  type        = string
+  description =  "The organization represents an unique identifier for a service owner and triggers the definition of groups on root compartment level"
+  default     = "Organization"
+  validation {
+    condition     = length(regexall("^[A-Za-z][A-Za-z]{1,26}$", var.organization)) > 0
+    error_message = "The service_name variable is required and must contain upto 15 alphanumeric characters only and start with a letter."
+  }
+}
+variable "solution"            { 
+  type        = string
+  description =  "The solution represents an unique identifier for a service defined on root compartment level"
+  default     = "Service"   # Define a name that identifies the service
+  validation {
+    condition     = length(regexall("^[A-Za-z][A-Za-z]{1,26}$", var.solution)) > 0
+    error_message = "The service_name variable is required and must contain alphanumeric characters only, start with a letter and 15 character max."
+  }
+}
+
+variable "repository" {
+  type        = string
+  description = "The service configuration is stored using infrastructure code in a repository"
+  default     = "https://github.com/torstenboettjer/ocloud-default-configuration"
+}
+
+variable "owner" {
+  type        = string
+  description = "The service owner is identified by his or her eMail address"
+  default     = "RobotNotExist@oracle.com"
```

Comment:
Can no email be provided by default here, requiring the user to provide an email input?

Reply:
Would throw an error. Use the same email dummy that we use on terraform github (publically)